### ISSUE TITLE
git dojo

### ISSUE THREAD

**Issue by RLesur** [spyrales/organisation#2]
Comme `git` devient de plus en plus utilisé dans nos métiers, que penseriez-vous d'un **git dojo** ? 

L'idée serait de découvrir, apprendre et réviser régulièrement les "routines" (katas ?) qu'on est amené à faire régulièrement : checkout, commit, amend, push, pull, les PR... 

Comme elles sont un peu différentes suivant l'environnement de travail, ça pourrait se faire une fois dans RStudio, une fois en ligne de commande, une fois dans GH Desktop, avec GitLab, GitHub, Gogs...

Je me dis que l'idéal serait d'avoir des vidéos (courtes) pour ça.

Qu'en pensez-vous ?

**Comment by gilles13** [comment]
ça me rendrait grandement service :+1: 

une solution pour "héberger" les videos ?

**Comment by strainel** [comment]
Oui c’est très utile, je galère régulièrement.

Pour les vidéos, on peut les mettre sous une chaîne YouTube. J’ai créer un compte Google hier pour récupérer les mails du formulaire de contact du futur site.

**Comment by Welsh44** [comment]
Un lien circule chez nous depuis quelques jours : https://learngitbranching.js.org/
J'ai commencé, j'ai bien aimé.
Mais oui pour les kata (avec ou sans S ?).

**Comment by RLesur** [comment]
@Welsh44 J'ai jeté un coup d'oeil. C'est super bien ! Un peu orienté développeurs quand même.  

Je dis ça car lorsque je forme des débutants (statisticiens) à git, je leur indique que s'ils croisent sur internet des commandes comme `git rebase` ou `git push --force`, il ne faut jamais les utiliser !  

Je suis assez directif sur cet aspect envers les débutants car je n'ai pas envie de rattraper des erreurs (qui pour certaines d'entre elles pourraient être impossibles à rattraper) !

**Comment by Welsh44** [comment]
@RLesur entièrement d'accord avec toi sur l'orientation développeurs.
On pourrait imaginer une version "git pour les nuls" et un deuxième niveau avec celui-ci.
Romain Warnan (Insee) a mis en ligne une formation (trop) riche : https://romain-warnan.github.io/formation-git/
Mais qui peut servir de base.

Pour les katas : https://github.com/romain-warnan/git-katas

**Comment by RLesur** [comment]
Oh ben dis donc, ce repo de katas est assez extraordinaire. C'est exactement ce que j'avais en tête ! J'aurais dû chercher un peu.

Je suis d'accord qu'il faudrait 2 niveaux que je nommerais volontiers :
- git sans danger
- Xtreme git

**Comment by strainel** [comment]
les katas sur git sont top ! 
à traduire peut être

**Comment by Welsh44** [comment]
Le repository initial est celui-ci (Romain l'a forké) : https://github.com/praqma-training/git-katas
On pourrait commencer par ne traduire que quelques uns qui nous semblent intéressants/utiles/indispensables.

**Comment by strainel** [comment]
yep ! faut trouver des traducteurs... à demander sur tchap ! :) je fais un message pour demander de l'aide.

**Comment by RLesur** [comment]
Pour la partie "git sans danger", j'ai oublié de mentionner une référence que j'aime beaucoup [Happy Git and GitHub for the useR](https://happygitwithr.com/) de Jenny Bryan ; on trouve les routines courantes dans la partie V. Pas sous forme de katas évidemment mais on n'en est pas très loin.

**Comment by jmbernabotto** [comment]
La plupart de vos post dans ce fil sont pour moi du chinois. Un "git pour les nuls" serait en effet le bienvenu. En comprendre la philosophie d'abord. Puis les 10 commandes indispensables pour collaborer. Moi c'est @RLesur qui m'a permis de me débrouiller un peu avec ce concept. 
Je suis sur Mac et j'ai téléchargé le GitHub Desktop qui a l'air d'en faire beaucoup sans ligne de commande. Non pas que je n'aime pas la ligne de commande (c'est comme cela que je garde à jour mon Anaconda avec des conda install ou avec pip pour installer des bibliothèques) mais cela peut être plus convivial avec une interface plutôt bien léchée.  Voilà, je vous livre mes premières impressions. En avouant que je suis à peu près certain que je vais prendre goût à cette collaboration.

**Comment by MaelTheuliere** [comment]
Hello, désolé je réagis tard (le confinement ne me rend pas très dispo en ce moment!) mais je trouve l'idée de git dojo très bonne. Le lien est intéressant mais il passe par la ligne de commande uniquement. Pour les habitué de Rstudio (pour R ou Python hein...), ce serait top la même chose mais intégré à l'interface, qu'en dites vous ? ça mettrait une petite marche en moins peut être...

**Comment by RLesur** [comment]
@MaelTheuliere Oui. Je pense que dans l'idéal, il faudrait tout montrer : la ligne de commande et les interfaces graphiques. Je suis d'accord que la présence d'une interface graphique a un côté "rassurant" pour les personnes qui débutent. Cependant, on a quand même besoin de passer souvent dans un terminal (même avec RStudio), c'est pour cela que je pense que la ligne de commande est indispensable pour git. De plus, pour des personnes habituées à des langages REPL comme R et python, je ne pense pas que la ligne de commande soit si difficile à appréhender.
